inst/include/model_simulation.hpp

```diff
@@ -131,5 +146,213 @@ void run_new_infections(int hiv_step,
   }
 }
 
+template<typename real_type>
+void run_art_progression_and_mortality(int hiv_step,
+                                       int time_step,
+                                       const Parameters<real_type> &pars,
+                                       const State<real_type> &state_curr,
+                                       State<real_type> &state_next,
+                                       IntermediateData<real_type> &intermediate) {
+  for (int g = 0; g < pars.num_genders; g++) {
+    for (int ha = 0; ha < pars.age_groups_hiv; ha++) {
+      for (int hm = intermediate.everARTelig_idx; hm < pars.disease_stages; hm++) {
+        for (int hu = 0; hu < pars.treatment_stages; hu++) {
+          intermediate.deaths_art =
+              pars.art_mortality(hu, hm, ha, g) * pars.artmx_timerr(hu, time_step) *
+              state_next.art_strat_adult(hu, hm, ha, g);
+          intermediate.hiv_deaths_age_sex(ha, g) += pars.dt * intermediate.deaths_art;
+          state_next.aids_deaths_art(hu, hm, ha, g) += pars.dt * intermediate.deaths_art;
+          intermediate.gradART(hu, hm, ha, g) = -intermediate.deaths_art;
+        }
+
+        for (int hu = 0; hu < (pars.treatment_stages - 1); hu++) {
+          intermediate.gradART(hu, hm, ha, g) +=
+              -state_next.art_strat_adult(hu, hm, ha, g) / pars.h_art_stage_dur(hu);
+          intermediate.gradART(hu + 1, hm, ha, g) +=
+              state_next.art_strat_adult(hu, hm, ha, g) / pars.h_art_stage_dur(hu);
+        }
+
+        // ART dropout
+        if (pars.art_dropout(time_step) > 0) {
+          for (int hu = 0; hu < pars.treatment_stages; hu++) {
+            intermediate.grad(hm, ha, g) += pars.art_dropout(time_step) * state_next.art_strat_adult(hu, hm, ha, g);
+            intermediate.gradART(hu, hm, ha, g) -=
+                pars.art_dropout(time_step) * state_next.art_strat_adult(hu, hm, ha, g);
+          }
+        }
+      }
+    }
+  }
+}
+
+template<typename real_type>
+void run_art_initiation(int hiv_step,
+                        int time_step,
+                        const Parameters<real_type> &pars,
+                        const State<real_type> &state_curr,
+                        State<real_type> &state_next,
+                        IntermediateData<real_type> &intermediate) {
+  for (int g = 0; g < pars.num_genders; ++g) {
+    intermediate.Xart_15plus = 0.0;
+    intermediate.Xartelig_15plus = 0.0;
+    intermediate.expect_mort_artelig15plus = 0.0;
+    for (int ha = pars.hIDX_15PLUS; ha < pars.age_groups_hiv; ++ha) {
+      for (int hm = intermediate.everARTelig_idx; hm < pars.disease_stages; ++hm) {
+        if (hm >= intermediate.anyelig_idx) {
+          // TODO: Implement special population ART eligibility
+          real_type prop_elig = 1.0;
+          intermediate.artelig_hahm(hm, ha - pars.hIDX_15PLUS) = prop_elig * state_next.hiv_strat_adult(hm, ha, g);
```

Comment:
do we have tickets to refactor some of these names (looking here at `hIDX_15PLUS` which stands out). It's possible that this is not so much a variable renaming issue though and a restructuring the data issue